crates/transaction-pool/src/traits.rs

```diff
@@ -372,6 +379,16 @@ impl PoolTransaction for PooledTransaction {
     fn size(&self) -> usize {
         self.transaction.transaction.input().len()
     }
+
+    /// Returns the transaction type
+    fn tx_type(&self) -> u8 {
+        self.transaction.tx_type().into()
+    }
+
+    /// Returns the length of the rlp encoded object
+    fn encoded_length(&self) -> usize {
+        self.transaction.length()
+    }
```

Comment:
this is sufficient for now,
perhaps we should think about adding this value to the `ValidatedTransaction` instead so we don't have to do this all the time when we send this transaction over p2p

Reply:
Yeah, I was just writing a comment about this, I wasn't able to access a `ValidPoolTransaction` while impl `encoded_length()`, so it must be something I don't understand.

Reply:
I think I get it. I'll wait this get merged and will do another PR.

Reply:
Yeah let's use the same pattern as in Transaction ECRECOVER where we don't want to recalculate the signer all the time because it's expensive so we put it in `TransactionEcRecovered` or whatever.